gadopt/gplates.py

```diff
@@ -0,0 +1,449 @@
+        times to the boundary conditions specified with dbc. Due to potential challenges in
+        identifying the plate id for a given point with pygplates, especially in high-resolution
+        simulations, this interface employs a method of calculating velocities at a number
+        of equidistant points on a sphere. It then interpolates these velocities for points
+        assigned a plate id. A warning is raised for any point not assigned a plate id.
+
+        Attributes:
+            rotation_filenames (str or list of str): Collection of rotation file names for pygplates.
+            topology_filenames (str or list of str): Collection of topology file names for pygplates.
+            geologic_zero (float): The oldest time present in the plate reconstruction model.
+            delta_time (float, optional): The time window range outside which plate velocities are updated. Defaults to 1.0.
+            scaling_factor (float, optional): Scaling factor for surface velocities. Defaults to 1.0.
+            nseeds (int, optional): Number of seed points used in the Fibonacci sphere generation. Defaults to 1000.
+            nneighbours (int, optional): Number of neighboring points to consider in velocity calculations. Defaults to 3.
+
+        Methods:
+            assign_plate_velocities(model_time):
+                Assigns plate velocities based on the specified model time.
+            ndtime2geotime(ndtime):
+                Converts non-dimensionalised time to geologic time.
+            geotime2ndtime(geotime):
+                Converts geologic time to non-dimensionalised time.
+
+        Examples:
+            >>> connector = pyGplatesConnector(rotation_filenames, topology_filenames, geologic_zero)
+            >>> connector.get_plate_velocities(model_time=100)
+        """
+
+        # Rotation model(s)
+        self.rotation_model = pygplates.RotationModel(rotation_filenames)
+
+        # Topological plate polygon feature(s).
+        self.topology_features = []
+        for fname in topology_filenames:
+            for f in pygplates.FeatureCollection(fname):
+                self.topology_features.append(f)
+        # geologic_zero is the same as model_time=0
+        self.geologic_zero = geologic_zero
+
+        # time velocity interpolations
+        self.delta_time = delta_time
+
+        # seeds are equidistantial points generate on a sphere
+        self.seeds = self._fibonacci_sphere(samples=int(nseeds))
+        # number of neighbouring points that will be used for interpolation
+
+        self.nneighbours = nneighbours
+        self.velocity_domain_features = (
+            self._make_GPML_velocity_feature(
+                self.seeds
+            )
+        )
+
+        # last reconstruction time
+        self.reconstruction_time = None
+
+        # Flag to know when to recalculate surface velocities.
+        self.recalculation_flg = False
```

Comment:
I'd suggest a name that suggests what the flag is signalling, maybe `self.needs_recalculation` or similar?